ansible/roles/infra-ec2-create-inventory/tasks/main.yml

```diff
@@ -46,7 +46,7 @@
   ignore_errors: yes
 
 - add_host:
-    name: "{{item.tags.internaldns | default(item.private_dns_name)}}"
+    name: "{{item.tags.internaldns if not install_cloud_provider|d(false)|bool else item.private_dns_name | default(item.private_dns_name)}}"
```

Comment:
We need to find another to defined this because `infra-ec2-create-inventory` is a core and common role, and `install_cloud_provider` is specific to your config.